src/containers/ManageListingsPage/ManageListingsPage.duck.js

```diff
@@ -281,6 +344,34 @@ export const openListing = listingId => (dispatch, getState, sdk) => {
     });
 };
 
+const delay = ms => new Promise(resolve => window.setTimeout(resolve, ms));
+export const discardDraft = listingId => (dispatch, getState, sdk) => {
+  dispatch(discardDraftRequest(listingId));
+
+  const { queryParams } = getState().ManageListingsPage;
+  const { perPage, ...rest } = queryParams;
+  const params = { ...rest, perPage };
+
+  return sdk.ownListings
+    .discardDraft({ id: listingId }, { expand: true })
+    .then(discardResponse => {
+      // Set delay before removing the draft so that the user
+      // notices which listing card gets removed
+      return delay(300).then(() => {
+        Promise.all([
+          dispatch(discardDraftSuccess(discardResponse)),
+          sdk.ownListings.query(params),
+        ]).then(([_, listingResponse]) => {
+          dispatch(addOwnEntities(listingResponse));
+          dispatch(queryListingsSuccess(listingResponse));
+        });
+      });
+    })
```

Comment:
It seems I understood your question about in-progress wrongly.

I was thinking a bit different flow:
```suggestion
  return sdk.ownListings
    .discardDraft({ id: listingId }, { expand: true })
    .then(discardResponse => {
      // Set delay before removing the draft so that the user
      // notices which listing card gets removed
      return Promise.all([
        delay(300), // ensure 300 ms delay if queryOwnListings is fast
        dispatch(queryOwnListings(queryParams)),
      ]);
    })
    .then(() => {
      dispatch(discardDraftSuccess(discardResponse)),
    })
    .catch(e =>{//...
```

Something like this
1. Call discardDraft
2. Then call promise.all for queryOwnListings with correct params and a delay
  2.1 Let existing queryOwnListings function to handle the query flow updates
  2.2 There seems to be useless reconstruction for the "queryParams" on the current implementation
  2.3 This can rely on [Promise chaining](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Using_promises#chaining).
3. Then mark success for the discard action
3.1 No need to call _removeDiscardedDraftListing_

Reply:
If there's need to return the discardResponse (but I don't think there's a need for that):

```js
  let discardRes = null;
  return sdk.ownListings
    .discardDraft({ id: listingId }, { expand: true })
    .then(discardResponse => {
      discardRes = discardResponse;
      // Set delay before removing the draft so that the user
      // notices which listing card gets removed
      return Promise.all([
        delay(300), // ensure 300 ms delay if queryOwnListings is fast
        dispatch(queryOwnListings(queryParams)),
      ]);
    })
    .then(() => {
      dispatch(discardDraftSuccess(discardResponse)),
      return discardResponse;
    })
    .catch(e =>{//...
    ```

Reply:
Logic fixed!